client/src/components/BuddyMatcher.jsx

```diff
@@ -51,9 +51,9 @@ const BuddyMatcher = () => {
             return;
         }
 
+        startSearch()
+        setLoadingText(defaultLoadingText)
```

Comment:
what's with this one?

Reply:
The first code was calling `socket.connect()` and `startNewSearch()` as a result, the `join` event is emitted twice;
* First, emit is immediately we call `startNewSearch()`:
https://github.com/Dun-sin/Whisper/blob/ae221c811648c747e5818546f7fff71850247e9e/client/src/components/BuddyMatcher.jsx#L32
* Second emit after the socket connects:
https://github.com/Dun-sin/Whisper/blob/ae221c811648c747e5818546f7fff71850247e9e/client/src/components/BuddyMatcher.jsx#L112

Line 54-55 is to restart the loading animation.

Reply:
It took me long enough to figure out my comment was not submitted. It was `PENDING`

Reply:
I get it, thanks for explaining